ddtrace/contrib/internal/ray/patch.py

```diff
@@ -479,20 +487,23 @@ def _(m):
     def _(m):
         _w(m.RemoteFunction, "_remote", traced_submit_task)
 
-    _w(ray, "wait", traced_wait)
+    if config.ray.trace_core_api:
+        _w(ray, "wait", traced_wait)
```

Comment:
you could always patch `traced_wait`, but bail early/not create a span if not `config.ray.trace_core_api`, this way `trace_core_api` becomes something that can be set at runtime instead of at startup/patch time only.

going with this approach might make testing easier too, since you can easily/temporarily enable the behavior of `trace_core_api`